errors.go

```diff
@@ -7,6 +7,13 @@ import (
 	"strings"
 )
 
+// httpErrorResponse allows to customize the format of non-200 http responses.
+// If the handler returns an error which implements this interface, it will be marshaled as is and written with the specified status code.
```

Comment:
Total nitpick: can we wrap at 78 lines like we do elsewhere?
```suggestion
// If the handler returns an error which implements HTTPErrorResponse, it will
// be marshaled as-is to the response and written with the specified status
// code.

```